src/configure.rs

```diff
@@ -260,10 +244,15 @@ pub fn write_configuration(doc: &Document) {
         .expect("Error writing starship config");
 }
 
-pub fn edit_configuration(editor_override: Option<&str>) -> Result<(), Box<dyn std::error::Error>> {
+pub fn edit_configuration(
+    context: &Context,
+    editor_override: Option<&str>,
+) -> Result<(), Box<dyn std::error::Error>> {
     // Argument currently only used for testing, but could be used to specify
     // an editor override on the command line.
-    let config_path = get_config_path();
+    let config_path = context
+        .get_config_path_os()
+        .expect("config path required to edit the config");
```

Comment:
This isn't really worse than what we had, but for user-facing errors like this, please prefer using `eprintln!` and `exit`, the output of `expect` can be hard to parse.